### PR TITLE
Feature/lap 17 renderer rework

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

File: legion/engine/core/common/managed_resource.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    template<typename T>
+    async::readonly_rw_spinlock managed_resource<T>::m_referenceLock;
+
+    template<typename T>
+    std::unordered_map<id_type, size_type> managed_resource<T>::m_references;
+
+    template<typename T>
+    std::atomic<id_type> managed_resource<T>::m_lastId;
+}
```
Comment (Algo-ryth-mix) [inline legion/engine/core/common/managed_resource.hpp]: there is not a single comment here

File: legion/engine/core/ecs/entity_handle.cpp [hunk 2]
```diff
@@ -155,6 +155,7 @@ namespace legion::core::ecs
         std::vector <ecs::entity_handle> children;
         oarchive(cereal::make_nvp("ID", m_id), cereal::make_nvp("NAME", std::string("ENTITY")));
         auto ent = m_registry->createEntity(m_id);
+        (void)ent;
```
Comment (Algo-ryth-mix) [inline legion/engine/core/ecs/entity_handle.cpp:158]: should probably be [[mabye_unused]] instead ?

File: legion/engine/rendering/data/buffer.hpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            glBindBuffer(m_target, m_id);
+        }
+
+        void release() const
+        {
+            glBindBuffer(m_target, 0);
+        }
+    };
+}
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/data/buffer.hpp]: needs a lot of comments

File: legion/engine/rendering/data/framebuffer.hpp [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        {
+            glBindFramebuffer(m_target, 0);
+        }
+
+        L_NODISCARD operator bool() const
+        {
+            return m_id;
+        }
+    };
+}
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/data/framebuffer.hpp]: also comments here please

File: legion/engine/rendering/data/renderbuffer.hpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            GLenum internalformat;
+            glGetRenderbufferParameteriv(GL_RENDERBUFFER, GL_RENDERBUFFER_INTERNAL_FORMAT, reinterpret_cast<GLint*>(&internalformat));
+            if (m_samples > 0)
+                glNamedRenderbufferStorageMultisample(m_id, m_samples, internalformat, newSize.x, newSize.y);
+            else
+                glNamedRenderbufferStorage(m_id, internalformat, newSize.x, newSize.y);
+            glBindRenderbuffer(GL_RENDERBUFFER, 0);
+        }
+
+    };
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/data/renderbuffer.hpp]: no comments, some manual formating required: lines exceed what github is able to show in one line

**Review by Algo-ryth-mix — APPROVED** [review]
overall more comments required, some lines need manual formating.
Try to keep pr shorter

**Review by tuhri444 — APPROVED** [review]

**Review by kukash — APPROVED** [review]

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]
see previous msgs

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**